pkg/cluster/deploystorage.go

```diff
@@ -302,3 +288,29 @@ func (m *manager) setMasterSubnetPolicies(ctx context.Context) error {
 
 	return m.subnet.CreateOrUpdate(ctx, m.doc.OpenShiftCluster.Properties.MasterProfile.SubnetID, s)
 }
+
+// generateInfraID take base and returns a ID that
+// - is of length maxLen
+// - contains randomLen random bytes
+// - only contains `alphanum` or `-`
+// see openshift/installer/pkg/asset/installconfig/clusterid.go for original implementation
+func generateInfraID(base string, maxLen int, randomLen int) string {
+	maxBaseLen := maxLen - (randomLen + 1)
```

Comment:
Looks good to me, just out of curiosity - why do we set a maximum?

Reply:
The rationale appears to be that you have a global maximum, and then a maximum of provided characters, so that you leave a few for randomness. Some clouds don't like long IDs, not sure if Azure is really one of them.